Makefile

```diff
@@ -39,7 +39,8 @@ endif
 GO                := go
 GO_MOD_ROOT       := go.opentelemetry.io/proto
 ALL_GO_MOD_DIRS := $(shell find . -type f -name 'go.mod' -exec dirname {} \; | sort)
-OTEL_GO_MOD_DIRS := $(filter-out $(TOOLS_MOD_DIR), $(ALL_GO_MOD_DIRS))
+ALL_GO_SUB_MOD_DIRS := $(shell find . -type f -name 'go.mod' -mindepth 2 -exec dirname {} \; | sort)
+OTEL_GO_MOD_DIRS := $(filter-out $(TOOLS_MOD_DIR), $(ALL_GO_SUB_MOD_DIRS))
```

Comment:
Do we still need `clean-gen` after these changes?

Reply:
hmm yeah, we could indeed remove it and only run tidy in the submodules. I'm thinking it's too easy to get bitten by gen code that hasn't been cleaned though. So doing it is a good measure as well.

Reply:
I would like to have `gen` (build artifacts) after running `make` to easier troubleshoot problems if something goes wrong.